### PR TITLE
Implement zypper-ps

### PR DESCRIPTION
**Pull request by flavio**
This PR introduces the zypper-ps command. I changed the `update` and `patch` commands to store the ID of the successfully updated/patched images to our cache. We later use these cached values to compare the images from which a container has been started.

While working on the feature I had to perform a couple of code refactorings to increase the code coverage level.

### CODE REVIEW

File: cache.go [hunk 1]
```diff
@@ -101,6 +104,19 @@ func (cd *cachedData) reset() {
 	cd.flush()
 }
 
+func (cd *cachedData) addImageToListOfUpdatedOnes(name string) error {
```
Comment (mssola) [inline cache.go]: Provide documentation for this.

File: ps.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	"github.com/codegangsta/cli"
+)
+
+// zypper-docker ps
+func psCmd(ctx *cli.Context) {
+	client := getDockerClient()
+	containers, err := client.ListContainers(false, false, "")
+	if err != nil {
+		log.Println("Cannot list running containers", err)
+		exitWithCode(1)
```
Comment (mssola) [inline ps.go]: return

**Comment by mssola** [comment]
LGTM. Some notes:

- As discussed on IRC, maybe rename `Updated` to `Outdated` or `Old`, or `Patched`, etc. to avoid confusion.
- When you have everything fixed, merge everything into a single commit and sign it.

**Comment by flavio** [comment]
Fixed, I renamed everything to "outdated".

**Pull request merged**